src/storage/mvcc/txn.rs

```diff
@@ -689,10 +689,34 @@ impl<S: Snapshot> MvccTxn<S> {
                     .reader
                     .get_txn_commit_info(&primary_key, self.start_ts)?
                 {
-                    Some((ts, write_type)) if write_type != WriteType::Rollback => {
-                        Ok((0, ts, false))
+                    Some((ts, write_type)) => {
+                        if write_type == WriteType::Rollback {
+                            Ok((0, 0, false))
+                        } else {
+                            Ok((0, ts, false))
+                        }
+                    }
+                    None => {
+                        let ts = self.start_ts;
```

Comment:
Can we extract a `rollback_nonexistent_lock`? It's a repetition of `cleanup`.

Reply:
Ok. I was thinking about simplifying it in another PR, but maybe I'd better do it right now.

Reply:
We can merge it first to let Jepsen pass.